be/src/vec/functions/function_jsonb.cpp

```diff
@@ -66,16 +66,85 @@
 
 namespace doris::vectorized {
 
-enum class NullalbeMode {
+enum class NullableMode {
     NULLABLE = 0,
     NOT_NULL,
     FOLLOW_INPUT,
 };
 
 enum class JsonbParseErrorMode { FAIL = 0, RETURN_NULL, RETURN_VALUE, RETURN_INVALID };
 
+inline static JsonbValue* find_value(const std::unique_ptr<JsonbWriter>& writer,
+                                     JsonbValue* param_json, JsonbPath& path, int path_idx,
+                                     hDictFind handler) {
+    JsonbValue* pval = param_json;
+    for (size_t i = path_idx; i < path.get_leg_vector_size(); ++i) {
+        switch (path.get_leg_from_leg_vector(i)->type) {
+        case MEMBER_CODE: {
+            if (LIKELY(pval->type() == JsonbType::T_Object)) {
+                if (path.get_leg_from_leg_vector(i)->leg_len == 1 &&
+                    *path.get_leg_from_leg_vector(i)->leg_ptr == WILDCARD) {
+                    continue;
+                }
+
+                pval = ((ObjectVal*)pval)
+                               ->find(path.get_leg_from_leg_vector(i)->leg_ptr,
+                                      path.get_leg_from_leg_vector(i)->leg_len, handler);
+
+                if (!pval) return nullptr;
+                continue;
+            } else {
+                return nullptr;
+            }
+        }
+        case ARRAY_CODE: {
+            if (path.get_leg_from_leg_vector(i)->leg_len == 1 &&
+                *path.get_leg_from_leg_vector(i)->leg_ptr == WILDCARD) {
+                if (LIKELY(pval->type() == JsonbType::T_Array)) {
+                    // make array value
+                    writer->writeStartArray();
+                    for (int j = 0; j < ((ArrayVal*)pval)->numElem(); ++j) {
+                        JsonbValue* v =
+                                find_value(writer, ((ArrayVal*)pval)->get(j), path, i + 1, handler);
+                        if (v) {
+                            writer->writeValue(v);
```

Comment:
value may be written by the same writer many times, since writer the the same one in recursive call